app/ai/voice/agents/breeze_buddy/agent/__init__.py

```
@@ -597,6 +622,42 @@ async def on_pipeline_error(task, error):
+            logger.warning(f"STT error detected from processor: {processor}")
+
+            # Record failure in fallback system (once per call, Soniox only)
+            if self.stt_provider == "soniox" and not self._stt_failure_recorded:
+                self._stt_failure_recorded = True
+                try:
+                    await record_stt_failure(
+                        error_msg=str(error_msg)[:200],
+                        call_sid=self.call_sid or "",
+                        context="mid-call",
+                    )
+                except Exception as fb_err:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)